Modules/Loadable/Markups/MRMLDM/vtkMRMLMarkupsDisplayableManager.cxx

```diff
@@ -88,18 +88,20 @@ vtkStandardNewMacro (vtkMRMLMarkupsDisplayableManager);
 //---------------------------------------------------------------------------
 vtkMRMLMarkupsDisplayableManager::vtkMRMLMarkupsDisplayableManager()
 {
-  this->Focus.insert("vtkMRMLMarkupsAngleNode");
-  this->Focus.insert("vtkMRMLMarkupsFiducialNode");
-  this->Focus.insert("vtkMRMLMarkupsLineNode");
-  this->Focus.insert("vtkMRMLMarkupsCurveNode");
-  this->Focus.insert("vtkMRMLMarkupsClosedCurveNode");
-  this->Focus.insert("vtkMRMLMarkupsPlaneNode");
+  this->RegisterMarkup(vtkMRMLMarkupsAngleNode::New(), vtkSlicerAngleWidget::New(), "A");
+  this->RegisterMarkup(vtkMRMLMarkupsFiducialNode::New(), vtkSlicerPointsWidget::New(), "F");
+  this->RegisterMarkup(vtkMRMLMarkupsLineNode::New(), vtkSlicerLineWidget::New(), "L");
+  this->RegisterMarkup(vtkMRMLMarkupsCurveNode::New(), vtkSlicerCurveWidget::New(), "C" );
+  this->RegisterMarkup(vtkMRMLMarkupsClosedCurveNode::New(), vtkSlicerClosedCurveWidget::New(), "C");
+  this->RegisterMarkup(vtkMRMLMarkupsPlaneNode::New(), vtkSlicerPlaneWidget::New(), "P");
```

Comment:
To me it would make more sense for each markup type to register itself so that this class wouldn't presume that any of them are available.  That would simplify maintenance since all would follow the same pattern.